src/main/c/NativeLongFunc.cpp

```diff
@@ -0,0 +1,142 @@
+//
+// src/main/c/NativeFunc.cpp - native funcs for Scala NativeLongFunc
+//
+// Richard Cownie, Hail Team, 2018-04-18
```

Comment:
we avoid author comments in the Scala code, I assume we'll do the same in C++ (cc: @cseed)